Kernel/Net/Socket.cpp

```diff
@@ -101,24 +101,27 @@ KResult Socket::queue_connection_from(NonnullRefPtr<Socket> peer)
     return KSuccess;
 }
 
-KResult Socket::setsockopt(int level, int option, const void* user_value, socklen_t user_value_size)
+KResult Socket::setsockopt(int level, int option, Userspace<const void*> user_value, socklen_t user_value_size)
 {
     ASSERT(level == SOL_SOCKET);
     switch (option) {
-    case SO_SNDTIMEO:
+    case SO_SNDTIMEO: {
         if (user_value_size != sizeof(timeval))
             return KResult(-EINVAL);
-        copy_from_user(&m_send_timeout, (const timeval*)user_value);
+        copy_from_user(&m_send_timeout, static_ptr_cast<const timeval*>(user_value));
         return KSuccess;
-    case SO_RCVTIMEO:
+    }
+    case SO_RCVTIMEO: {
```

Comment:
The braces were from a previous now rebased commit where there were locals introduced, I'll remove them.